src/database.c

```diff
@@ -115,16 +123,39 @@ log_database_init(ProfAccount* account)
         goto out;
     }
 
-    query = "CREATE TABLE IF NOT EXISTS `DbVersion` ( `dv_id` INTEGER PRIMARY KEY, `version` INTEGER UNIQUE)";
+    gboolean database_exists = (_get_db_version() != -1);
+
+    query = "CREATE TABLE IF NOT EXISTS `DbVersion` (`dv_id` INTEGER PRIMARY KEY, `version` INTEGER UNIQUE)";
     if (SQLITE_OK != sqlite3_exec(g_chatlog_database, query, NULL, 0, &err_msg)) {
         goto out;
     }
 
-    query = "INSERT OR IGNORE INTO `DbVersion` (`version`) VALUES('1')";
+    query = "INSERT OR IGNORE INTO `DbVersion` (`version`) VALUES ('1')";
     if (SQLITE_OK != sqlite3_exec(g_chatlog_database, query, NULL, 0, &err_msg)) {
         goto out;
     }
 
+    // Check && Update DB based on version
+
+    int db_version = _get_db_version();
+    if (db_version == -1) {
+        cons_show_error("DB Initialization Error: Unable to check DB version.");
+        goto out;
+    }
+
+    if (db_version < 2) {
+        if (database_exists) {
+            cons_show("Updating DB to V2. Please, wait...");
```

Comment:
User shouldn't be bothered with too much information and text. This should go as seemless as possible. I would maybe write `Updating Database schema to new version` or something like that. So that we can leave this text also vor a V3 and so on.

Reply:
Generally lets write `database` and not `DB` to the user. Also in the messages below.

Reply:
> User shouldn't be bothered with too much information and text. This should go as seemless as possible. I would maybe write `Updating Database schema to new version` or something like that. So that we can leave this text also vor a V3 and so on.

version 2 is good since we know which exact version update was problematic (e.g. it can stuck).

Reply:
I think the user just needs to know that the DB is migrated. In the log file he can find the information from which version to which it is happening in case of an error. So I would probably write Migrate instead of Update here too.

Reply:
```c
            cons_show("Migrating database schema. This operation may take a while...");
```
But I want to keep this message to avoid user's panic if program is stuck for (e.g.) 5-50 seconds.